crypto/bio/b_sock.c

```diff
@@ -47,7 +47,10 @@ int BIO_get_host_ip(const char *str, unsigned char *ip)
             BIO_ADDR_rawaddress(BIO_ADDRINFO_address(res), NULL, &l);
             /* Because only AF_INET addresses will reach this far,
                we can assert that l should be 4 */
-            OPENSSL_assert(l == 4);
+            if (!ossl_assert(l == 4)) {
+                BIO_ADDRINFO_free(res);
+                return 0;
+            }
 
             BIO_ADDR_rawaddress(BIO_ADDRINFO_address(res), ip, &l);
             ret = 1;
```

Comment:
Given that there is another call BIO_ADDRINFO_free(res) I'd say that following would be more readable:
```
if (oss_assert(l == 4) {
    BIO_ADDR_rawaddress(BIO_ADDRINFO_address(res), ip, &l);
    ret = 1;
}
```
But this is actually insufficient. BIO_ADD_rawaddress actually can fail returning 0, and then `l` would be uninitialized and happen to be four. So that correct solution ought to be
```
if (oss_assert(BIO_ADDR_rawaddress(BIO_ADDRINFO_address(res), NULL, &l) && l == 4)
    ret = BIO_ADDR_rawaddress(BIO_ADDRINFO_address(res), ip, &l);
```